src/angular/hq/src/app/staff-dashboard/staff-dashboard-time-entry/staff-dashboard-time-entry.component.ts

```diff
@@ -393,4 +394,16 @@ export class StaffDashboardTimeEntryComponent
       await this.modalService.alert('Rejection', this.time.rejectionNotes);
     }
   }
+
+  private setMaximumHours(): void {
+    const maxTimeEntry = this.time?.maximumTimeEntryHours;
+    if (maxTimeEntry !== undefined && maxTimeEntry !== null) {
+      this.form.controls.hours.setValidators([
+        Validators.required,
+        Validators.min(0.25),
+        Validators.max(maxTimeEntry),
+      ]);
+      this.form.controls.hours.updateValueAndValidity();
+    }
```

Comment:
This logic looks good but we should really be keying off of the `chargeCodeId` changing instead of the `time` input value so it handles both when creating a new time entry (without an existing time input) and an when existing time entries charge code changes.